src/reader.rs

```diff
@@ -3786,20 +3808,28 @@ impl<'a> Reader<'a> {
                 if !SCROLL_FORWARD_SUPPORTED.load() || !CURSOR_UP_SUPPORTED.load() {
                     return;
                 }
-                let Some(wait) = self.blocking_wait() else {
-                    self.request_cursor_position(
-                        &mut Outputter::stdoutput().borrow_mut(),
-                        Some(CursorPositionWait::ScrollbackPush),
-                    );
-                    return;
-                };
-                match wait {
-                    BlockingWait::Startup(_) => panic!(),
-                    BlockingWait::CursorPosition(_) => {
+                match self.blocking_wait() {
```

Comment:
(this panic seemed correct too; we do not execute any readline command until
we're either done waiting for startup response or we have canceled it via ctrl-c or timeout)